crates/primitives/src/transaction/mod.rs

```diff
@@ -1330,12 +1330,18 @@ impl TransactionSigned {
         }
 
         // Check if the tx is a list
-        if data[0] >= EMPTY_LIST_CODE {
+        let result = if data[0] >= EMPTY_LIST_CODE {
             // decode as legacy transaction
             TransactionSigned::decode_rlp_legacy_transaction(data)
         } else {
             TransactionSigned::decode_enveloped_typed_transaction(data)
+        };
```

Comment:
can we do `?` before doing the length check,

I assume we won't fully decode if there's an error so if result is an `Err` then we'd return `RlpError::UnexpectedLength` instead of the error?

Reply:
good catch! done here https://github.com/paradigmxyz/reth/pull/8296/commits/ebbf27bee91169f87b0e448579d0173ebceb11f6 PTAL